packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -620,6 +621,238 @@ abstract class DialogWindowController extends BaseWindowController {
+///     ];
+///
+///     if (_tooltipController != null) {
+///       children.add(
+///         TooltipWindow(
+///           controller: _tooltipController!,
+///           child: Container(
+///             padding: const EdgeInsets.all(8),
+///             color: Colors.black,
+///             child: const Text('This is a tooltip', style: TextStyle(color: Colors.white)),
+///           ),
+///         ),
+///       );
+///     }
+///
+///     return MouseRegion(
+///       onEnter: (_) => setState(
+///         () => _tooltipController = TooltipWindowController(
+///           parent: WindowScope.of(context),
+///           anchorRect: _getAnchorRect()!,
+///           positioner: const WindowPositioner(
+///             parentAnchor: WindowPositionerAnchor.right,
+///             childAnchor: WindowPositionerAnchor.left,
+///           ),
+///         ),
+///       ),
+///       onExit: (_) => setState(() {
+///         _tooltipController?.destroy();
+///         _tooltipController = null;
+///       }),
+///       cursor: SystemMouseCursors.click,
+///       child: AnimatedContainer(
+///         duration: const Duration(milliseconds: 200),
+///         color: _tooltipController != null ? Colors.blueAccent : Colors.blue,
+///         padding: const EdgeInsets.all(12),
+///         child: Row(children: children),
+///       ),
+///     );
+///   }
+///
+///   Rect? _getAnchorRect() {
+///     final RenderBox? renderBox = _key.currentContext?.findRenderObject() as RenderBox?;
+///     if (renderBox != null) {
+///       final Offset position = renderBox.localToGlobal(Offset.zero);
+///       final Size size = renderBox.size;
+///       return position & size; // creates a Rect
+///     }
+///
+///     return null;
+///   }
+/// }
+/// ```
+/// {@end-tool}
+///
+/// Children of a [TooltipWindow] widget can access the [TooltipWindowController]
+/// via the [WindowScope] inherited widget.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+abstract class TooltipWindowController extends BaseWindowController {
+  /// Creates a [TooltipWindowController] with the provided properties.
+  ///
+  /// Upon construction, the window is created by the platform.
+  ///
+  /// The [parent] argument specifies the parent window of this tooltip.
+  ///
+  /// The [anchorRect] argument specifies the rectangle in the parent's coordinate
+  /// space to which the tooltip is anchored.
+  ///
+  /// The [positioner] argument specifies how the tooltip should be positioned
+  /// relative to the [anchorRect].
+  ///
+  /// {@macro flutter.widgets.windowing.constraints}
+  ///
+  /// The [delegate] argument can be used to listen to the window's
+  /// lifecycle. For example, it can be used to save state before
+  /// a window is closed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  factory TooltipWindowController({
+    required BaseWindowController parent,
+    required Rect anchorRect,
+    required WindowPositioner positioner,
```

Comment:
Could/should this be optional? While I expect most people will want to provide a `positioner`, could we have a sensible default? I don't feel strongly, just seeing what people think.

Reply:
Thanks for confirming! Marking resolved